### PR TITLE
Improve source code organization in `palace/` directory

### PR DESCRIPTION
**Pull request by sebastiangrimberg**
Improves organization of source code files between the `drivers`, `fem`, `models`, `linalg`, and `utils` directories.

Also makes include guards specific to subdirectory in the future case of files with identical names in different directories.

### CODE REVIEW

File: palace/linalg/chebyshev.hpp [hunk 1]
```diff
@@ -1,8 +1,8 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-#ifndef PALACE_CHEBYSHEV_SMOOTHER_HPP
-#define PALACE_CHEBYSHEV_SMOOTHER_HPP
+#ifndef PALACE_LINALG_CHEBYSHEV_SMOOTHER_HPP
+#define PALACE_LINALG_CHEBYSHEV_SMOOTHER_HPP
```
Comment (hughcars) [inline palace/linalg/chebyshev.hpp:5]: Could drop the `SMOOTHER`, doesn't match the file name unlike all the others.

File: palace/linalg/distrelaxation.hpp [hunk 2]
```diff
@@ -1,8 +1,8 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-#ifndef PALACE_DIST_RELAXATION_SMOOTHER_HPP
-#define PALACE_DIST_RELAXATION_SMOOTHER_HPP
+#ifndef PALACE_LINALG_DIST_RELAXATION_SMOOTHER_HPP
+#define PALACE_LINALG_DIST_RELAXATION_SMOOTHER_HPP
```
Comment (hughcars) [inline palace/linalg/distrelaxation.hpp:5]: As with Chebyshev, should match the file name.

File: palace/linalg/pc.hpp [hunk 3]
```diff
@@ -1,8 +1,8 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-#ifndef PALACE_KSP_PRECONDITIONER_HPP
-#define PALACE_KSP_PRECONDITIONER_HPP
+#ifndef PALACE_LINALG_KSP_PRECONDITIONER_HPP
+#define PALACE_LINALG_KSP_PRECONDITIONER_HPP
```
Comment (hughcars) [inline palace/linalg/pc.hpp:5]: The KSP doesn't match the file name, and if I understand the acronym correctly, this reads as "Krylov Subspace Preconditioner Preconditioner" which is a bit ATM machine.

File: palace/models/surfaceconductivityoperator.hpp [hunk 4]
```diff
@@ -1,8 +1,8 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-#ifndef PALACE_SURF_CONDUCTIVITY_OPERATOR_HPP
-#define PALACE_SURF_CONDUCTIVITY_OPERATOR_HPP
+#ifndef PALACE_MODELS_SURF_CONDUCTIVITY_OPERATOR_HPP
+#define PALACE_MODELS_SURF_CONDUCTIVITY_OPERATOR_HPP
```
Comment (hughcars) [inline palace/models/surfaceconductivityoperator.hpp:5]: Surf should be Surface, the shortening is inconsistent with other head blockers. This holds for all the SURF_ in the other header blockers too.

File: palace/utils/configfile.hpp [hunk 5]
```diff
@@ -1,8 +1,8 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-#ifndef PALACE_CONFIGFILE_HPP
-#define PALACE_CONFIGFILE_HPP
+#ifndef PALACE_UTILS_CONFIGFILE_HPP
+#define PALACE_UTILS_CONFIGFILE_HPP
```
Comment (hughcars) [inline palace/utils/configfile.hpp:5]: Is configfile one word?

**Review by hughcars — COMMENTED** [review]
Wasn't totally in scope of the PR, but given all the header blockers are being touched, there's a few additional change I propose.

File: palace/linalg/chebyshev.hpp (hunk 1 above)
Comment (sebastiangrimberg) [reply palace/linalg/chebyshev.hpp:5]: The include guards match the class name, not file name in this case.

File: palace/linalg/pc.hpp (hunk 3 above)
Comment (sebastiangrimberg) [reply palace/linalg/pc.hpp:5]: As above.

Comment (sebastiangrimberg) [reply palace/linalg/pc.hpp:5]: As above, the include guard matches the `KspPreconditioner` class name. KSP is a PETSc acronym for Krylov subspace solvers, the "P" is not "Preconditioner" (might be better described as PKS solver, preconditioned Krylov subspace solver, or maybe the SP is for SubsPace, either way not our problem here).

File: palace/utils/configfile.hpp (hunk 5 above)
Comment (sebastiangrimberg) [reply palace/utils/configfile.hpp]: Addressed in ec816d9.

File: palace/models/surfaceconductivityoperator.hpp (hunk 4 above)
Comment (sebastiangrimberg) [reply palace/models/surfaceconductivityoperator.hpp]: Addressed in [ec816d9](https://github.com/awslabs/palace/pull/46/commits/ec816d9b22aee3d31f7bae3f54677f06374eff06).

**Review by hughcars — APPROVED** [review]

**Pull request merged**